### PR TITLE
Add testsuite integration and address incorrect handling of several instructions.

### PR DESCRIPTION
**Pull request by abarisani**
I identified several issues by executing known firmware code under ARMv6m_Simulator.

In order to debug them I thought necessary to add a proper testsuite, for this reason I pulled several (but not all) tests from the https://github.com/adamgreen/pinkySim repository.

After integrating the testsuite I began addressing each individual failing case, which lead to restructuring several instruction handling closely to original ARM specifications, with help from their pseudocode as well as pinkySim implementation.

After fixing several critical errors now the behavior of my known firmware reference is consistent with expectations as well as behavior under qemu.

Some tests pulled in from pinkySim still are not passing (I've left them commented in tests/main.cpp), I will hopefully address those in the future but I can say that so far they don't affect my known firmware execution. However this doesn't exclude that the instruction implementation still contains bugs.

Again I hope to make the entire testsuite passing, for now this pull request definitely improves the correctness of the simulation.

I also took the liberty of cleaning up Debug object files and binaries which I don't believe are fit for the git repository and are probably now obsolete due to the implemented testsuite.

Thanks!

### CODE REVIEW

File: src/Instruction.cpp [hunk 1]
```diff
@@ -38,34 +42,47 @@ uint8_t Instruction::getZERO(uint32_t inst)
...
+
+    if (shift != 0)
     {
-        if(getMSB(op1) != getMSB(result))
-            return 1;
-        else
-            return 0;
+        switch (type)
+        {
+        case LSL:
```
Comment (whdlgp) [inline src/Instruction.cpp:67]:
```cpp
value & (1 << (32 - shift));
(value & (1 << (shift - 1)));
reg->PSR[psrC] = result & (1 << 31);
```

I think, these might be exceed size of PSR[psrC], decleared with uint8_t .

```cpp
    uint32_t R[REG_SIZE];
    uint8_t PSR[PSR_SIZE];
```

The PSR should be either 0 or 1 (to represent false or true). 
in LSL, if value is 

1110_0110_0011_0000_1111_1001_0110_0011

and shift is 3, than result of "value & (1 << (32 - shift))" is 

0010_0000_0000_0000_0000_0000_0000_0000

I think, PSR[psrC] should store '1', not 0x20000000.

Was this result intentional? please check it before I merge this 
if not intentional, than I will fix and merge it

**Review by whdlgp — COMMENTED** [review]
I added a review of commit e90d43f, please check it.

Other things looks nice. 

In fact, this simulator was made roughly a week or so, and was left untouched, so it is true that the debugging environment is not equipped.

If this is reflected, it will be a good measure to check the accuracy of the simulator.

good pull request. Thanks.

File: src/Instruction.cpp (hunk 1 above)
Comment (abarisani) [reply src/Instruction.cpp:67]: I'll take a look and get back to you in the next days. Thanks.

Comment (abarisani) [reply src/Instruction.cpp:67]: This should be improved in https://github.com/whdlgp/ARMv6m_Simulator/pull/5/commits/e58d030599fc1f9d68e0ebe006034f5c6200f4dd

I uncommented more passing tests, ROR needs more work and I can see to it later on. For now this should address your concern. Thanks.

**Pull request merged**